src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -1179,6 +1179,15 @@ public StringBuilder Append(char value)
                 return this;
             }
 
+            unsafe
+            {
+                Span<char> tempBuffer = stackalloc char[32]; // should be enough for all primitives
```

Comment:
Wouldn't it be better to have this implemented as few specialized methods instead of a generic method which codegen won't be able to share between primitive types?

Reply:
Why write each specialized method out by hand rather than letting generic specialization take care of it?

Reply:
Which generic specialization? I doubt any of our generic specializations is smart enough to fold byte/ushort/uint cases into a simple TryUInt32ToDecStr call and remove the extra arguments.

Reply:
I meant when using the public APIs, that there's no reason to manually write out a version that calls uint.TryFormat, a version that calls byte.TryFormat, etc.  Unless there's an impactful performance boost from having multiple implementations to maintain and using internal APIs, I prefer the simpler one-method-handles-it-all approach.